epoxy-processor/src/main/java/com/airbnb/epoxy/processor/ModelBuilderInterfaceWriter.kt

```diff
@@ -74,6 +74,14 @@ class ModelBuilderInterfaceWriter(
             addTypeVariables(modelInfo.typeVariables)
             addMethods(interfaceMethods)
 
+            for (implements in modelInfo.superClassElement.interfaces) {
+                if (implements.toString() == ClassNames.MODEL_COLLECTOR.toString()) {
```

Comment:
it would be nice to memoize `ClassNames.MODEL_COLLECTOR.toString()`, or better yet, memoize on the class level via `memoizer.implementsModelCollector(modelInfo.superClassElement)`

The `memoizer` is a shared instance of Memoizer in the processor module that can be passed into this class

Reply:
Done. 

Are you worried about performance in general? Theses was simple `.toString()` and iteration on a likely small collections.

Reply:
@eboudrant thanks! I am a bit worried about performance - I recently reworked a lot of code to memoize hot spots and it improved performance several times (we dropped processing in one module from 4 seconds to half a second).

The `toString` functions here are a bit deceiving, one is in the ClassName class, which calls `emit` to a code writer to format the code for display which isn't trivial. It is internally cached, which I forgot about, but that isn't synchronized, which we have to worry about now since I added parallel processing support. The interface `toString` function is on the javac class `ClassType` which potentially loads the class by reading it from file which is not thread safe, so the memoizer helps us to synchronize.